services/packagist/packagist-php-version.service.js

```diff
@@ -68,17 +68,68 @@ module.exports = class PackagistPhpVersion extends BasePackagistService {
     }
   }
 
-  transform({ json, user, repo, version = '' }) {
-    const packageVersion =
-      version === ''
-        ? this.getDefaultBranch(json, user, repo)
-        : json.packages[this.getPackageName(user, repo)][version]
+  findVersionIndex(json, version) {
+    return json.findIndex(v => v.version === version)
+  }
+
+  async findSpecifiedVersion(json, user, repo, version, server) {
+    let release
+
+    if ((release = json[this.findVersionIndex(json, version)])) {
+      return release
+    } else {
+      try {
+        const allData = await this.fetchDev({
+          user,
+          repo,
+          schema: allVersionsSchema,
+          server,
+        })
+
+        const decompressed = this.decompressResponse(
+          allData,
+          this.getPackageName(user, repo)
+        )
+
+        return decompressed[this.findVersionIndex(decompressed, version)]
+      } catch (e) {
+        return release
+      }
+    }
+  }
+
+  async transform({ json, user, repo, version = '', server }) {
```

Comment:
The logic path for this badge is a bit muddled
Our sequence of execution is fetch --> transform --> render (or sometimes just fetch --> render)

- Fetch gets data from APIs
- Transform re-shapes the data (if necessary)
- Render turns it into a badge

Here we are making an API call in transform, so that's a good indicator we've split the logic up incorrectly or named things poorly.

In this situation, calling `fetch()`, trying to find the latest version in there, falling back to calling `fetchDev()` if necessary is all part of the "fetch" stage.

The "transform" stage (to the extent there is one for this badge) is just extracting `.require.php`